fs/fs.c

```diff
@@ -47,26 +47,10 @@ int32_t pi_fs_mount(struct pi_device *device)
 {
   struct pi_fs_conf *conf = (struct pi_fs_conf *)device->config;
   pi_fs_api_t *api = conf->api;
-  
+
   if (api == NULL)
   {
-    switch (conf->type)
-    {
```

Comment:
I think this wll break all the tests and applications which is using pi_fs_conf_init, did you check ?
If so, I'm anyway not sure if we can find a compatible solution.